### PR TITLE
update README to add coreos support and kustomize required

### PR DESCRIPTION
**Pull request by hchenxa**
**What this PR does / why we need it**:

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

**Special notes for your reviewer**:

1. Please confirm that if this PR changes any image versions, then that's the sole change this PR makes.

Update the document to add coreos support and kustimize package required.

**Release note**:

```release-note

```

### CODE REVIEW

File: cmd/clusterctl/examples/openstack/README.md [hunk 1]
```diff
@@ -18,31 +18,36 @@
...
+2. Install `kustomize` v1.0.11 which can be found [here](https://github.com/kubernetes-sigs/kustomize/releases/tag/v1.0.11).
 
 ## Generation
 For convenience, a generation script which populates templates based on openstack cloud provider
 configuration is provided.
 
 1. Run the generation script.
 ```
-./generate-yaml.sh --provider-os [os name]
+./generate-yaml.sh [options] <path/to/clouds.yaml> openstack --provider-os [os name]
```
Comment (jichenjc) [inline cmd/clusterctl/examples/openstack/README.md]: generate-yaml.sh [options] <path/to/clouds.yaml> <cloud> <provider os: [centos,ubuntu,coreos]>


this is current grammar

Comment (hchenxa) [reply cmd/clusterctl/examples/openstack/README.md]: seems we need to provide the os cloud, refer below code:

```
# Check if os cloud is provided
if [[ -n "$2" ]] && [[ $2 != -* ]] && [[ $2 != --* ]]; then
  CLOUD=$2
else
  echo "Error: No cloud specified"
  echo "You mush specify which cloud you want to use."
  echo ""
  print_help
  exit 1
fi
```

Comment (hchenxa) [reply cmd/clusterctl/examples/openstack/README.md]: I will update the [os name] to [[centos,ubuntu,coreos]]

File: cmd/clusterctl/examples/openstack/README.md [hunk 2]
```diff
@@ -18,31 +18,51 @@
...
+2. Install `kustomize` v1.0.11 which can be found [here](https://github.com/kubernetes-sigs/kustomize/releases/tag/v1.0.11).
 
 ## Generation
 For convenience, a generation script which populates templates based on openstack cloud provider
 configuration is provided.
 
 1. Run the generation script.
 ```
-./generate-yaml.sh --provider-os [os name]
+./generate-yaml.sh [options] <path/to/clouds.yaml> <openstack cloud> --provider-os [centos,ubuntu,coreos]
```
Comment (jichenjc) [inline cmd/clusterctl/examples/openstack/README.md]: ./generate-yaml.sh  clouds.yaml openstack --provideros coreos
Error: No provider OS specified
You must choose between the following operating systems: centos, ubuntu, coreos


root@kvm-017751:~/go/src/sigs.k8s.io/cluster-api-provider-openstack/cmd/clusterctl/examples/openstack# ./generate-yaml.sh  clouds.yaml openstack coreos
root@kvm-017751:~/go/src/sigs.k8s.io/cluster-api-provider-openstack/cmd/clusterctl/examples/openstack#

Comment (hchenxa) [reply cmd/clusterctl/examples/openstack/README.md]: yes, replaced the `--provider-os` to `<provider os>`.

**Pull request merged**